pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -397,7 +398,7 @@ impl DeltaLayer {
         end_lsn: Lsn,
         dropped: bool,
         predecessor: Option<Arc<dyn Layer>>,
-        page_versions: BTreeMap<(u32, Lsn), PageVersion>,
+        page_versions: impl Iterator<Item = (&'a (u32, Lsn), &'a PageVersion)>,
```

Comment:
This lifetime can be defined directly in function signature: `create<'a>` since it is used only in that function for parameters and do not interact with other impl members